### PR TITLE
Refactored common elements of GenerateAssertionsRequest.java and GetA…

### PR DESCRIPTION
**Pull request by vteague**
…ssertionsRequest.java into an abstract superclass called ContestRequest.java

Completes #51

### CODE REVIEW

File: src/main/java/au/org/democracydevelopers/raireservice/request/GenerateAssertionsRequest.java [hunk 1]
```diff
@@ -13,22 +13,13 @@ the raire assertion generation engine (https://github.com/DemocracyDevelopers/ra
 
 import au.org.democracydevelopers.raireservice.persistence.repository.ContestRepository;
 import java.util.List;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import org.springframework.data.annotation.ReadOnlyProperty;
```
Comment (michelleblom) [inline src/main/java/au/org/democracydevelopers/raireservice/request/GenerateAssertionsRequest.java:17]: I'm still not sure what it means for a request to "describe a contest".

File: src/main/java/au/org/democracydevelopers/raireservice/request/ContestRequest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
+/*
```
Comment (michelleblom) [inline src/main/java/au/org/democracydevelopers/raireservice/request/ContestRequest.java:1]: Do you want to copy the template I have used in the CVR database retrieval branch? It has some line breaks and conforms to the line width limit.

**Review by michelleblom — APPROVED** [review]
I have left some commenting requests, but these can be done in the InputVerification branch before merging into main.

File: src/main/java/au/org/democracydevelopers/raireservice/request/ContestRequest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+*/
+
+package au.org.democracydevelopers.raireservice.request;
+
+import au.org.democracydevelopers.raireservice.persistence.repository.ContestRepository;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.annotation.ReadOnlyProperty;
+
```
[the remark this answers is not in the document]
Comment (vteague) [reply src/main/java/au/org/democracydevelopers/raireservice/request/ContestRequest.java:19]: I changed it to 'identifying a contest and listing its candidates' - its an abstract class that summarises only those things that are common to all requests. Maybe I haven't made it very clear - feel free to suggest better wording.

File: src/main/java/au/org/democracydevelopers/raireservice/request/GenerateAssertionsRequest.java (hunk 1 above)
Comment (vteague) [reply src/main/java/au/org/democracydevelopers/raireservice/request/GenerateAssertionsRequest.java:17]: Changed to "identifying the contest..."

**Pull request merged**